### PR TITLE
Fix assigning from complex to float 

### PR DESCRIPTION
**Pull request by okuta**
This PR resolves #1316

### CODE REVIEW

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
@unnonouno Cloud you review this?

File: tests/cupy_tests/core_tests/test_ndarray_indexing.py [hunk 1]
```diff
@@ -137,6 +139,16 @@ def test_setitem_copy(self, xp, dtype):
         a[:] = b
         return a
 
+    @testing.for_all_dtypes_combination(('src_type', 'dst_type'))
+    @testing.numpy_cupy_array_equal()
+    def test_setitem_different_type(self, xp, src_type, dst_type):
+        a = xp.zeros((2, 3, 4), dtype=dst_type)
+        b = testing.shaped_arange((2, 3, 4), xp, src_type)
+        with numpy.warnings.catch_warnings():
```
Comment (toslunar) [inline tests/cupy_tests/core_tests/test_ndarray_indexing.py]: Import `warnings` instead.
```
>>> import numpy
>>> import warnings
>>> numpy.warnings is warnings
True
```

Comment (okuta) [reply tests/cupy_tests/core_tests/test_ndarray_indexing.py]: ok

File: tests/cupy_tests/core_tests/test_ndarray_indexing.py [hunk 2]
```diff
@@ -137,6 +140,16 @@ def test_setitem_copy(self, xp, dtype):
         a[:] = b
         return a
 
+    @testing.for_all_dtypes_combination(('src_type', 'dst_type'))
+    @testing.numpy_cupy_array_equal()
+    def test_setitem_different_type(self, xp, src_type, dst_type):
+        a = xp.zeros((2, 3, 4), dtype=dst_type)
+        b = testing.shaped_arange((2, 3, 4), xp, src_type)
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore')
```
Comment (toslunar) [inline tests/cupy_tests/core_tests/test_ndarray_indexing.py]: Why don't you specify the warning class?
```suggestion
            warnings.simplefilter('ignore', numpy.ComplexWarning)
```

Comment (okuta) [reply tests/cupy_tests/core_tests/test_ndarray_indexing.py]: Thanks, I fixed.

**Comment by toslunar** [comment]
Could you resolve the conflicts?
`elementwise_copy` has been moved to `cupy/core/_ufuncs.py`.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by okuta** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
The Jenkins failures seem unrelated to the PR.

**Pull request merged**